src/librustc_typeck/check/mod.rs

```diff
@@ -2418,6 +2418,14 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                             if arg_types.len() == 1 {""} else {"s"},
                             args.len(),
                             if args.len() == 1 {" was"} else {"s were"});
+                        let input_types = fn_inputs.iter().map(|i| format!("{:?}", i))
+                            .collect::<Vec<String>>();
+                        if input_types.len() > 0 {
+                            tcx.sess.span_note_without_error(sp,
+                                &format!("the following parameter type{} expected: {}",
+                                    if expected_arg_count == 1 {" was"} else {"s were"},
+                                    input_types.join(", ")));
+                        }
```

Comment:
This should probably be in its own function to reduce code repetition (this would allow `E0057`, `E0060` and `E0061` to be merged into a single error number).

Also it should be using the error builders, so you can attach the note to the main error.